assets/javascript/app.js

```diff
@@ -57,18 +58,27 @@ $(document).ready(function() {
             var state = $('#address-state').val()
             var zip = $('#address-zip').val()
 
-            if (address1 === '' && city === '' && state === '' && zip === '' && distance.toString() === $('#distance').val()) { 
+            // No address distance same, reject update
+            if (address1 === '' && city === '' && state === '' && zip === '' && distance.toString() === $('#distance').val()) {
                 // Return error that relevant information is already there 
                 $('.error').css('display','block')
                 $('.error').html('Results within a ' + distance + ' mile radius are displayed. Please change the radius or add an address.')
+
+            // No address distance different, update distance
             } else if (address1 === '' && city === '' && state === '' && zip === '' && distance.toString() !== $('#distance').val()) { 
                 // Do not change coordinates, but run localBrews() with new distance
                 distance = $('#distance').val()
                 localBrews()
-            } else if (address1 === '' && city === '' && state === '' && zip === '') {
+
+            // If address only reject
+            } else if (address1 !== '' && city === '' && state === '' && zip === '') {
                 $('.error').css('display','block')
                 $('.error').html('Please add city, state or zip')
-            } else if ((address1 === '' && (city !== '' || state !== '' || zip !== '') || (address1 !== '' || city !== '' || state !== '' || zip !== ''))) {
+
+            } else {
+                // On filter submit force scroll to top so new map can be seen
+                $('html, body').animate({scrollTop:0},0)
```

Comment:
If no error is returned on the filter, push the user to the top of the page so they can see the refreshed map and location. User filter=true on L99 as a switch for determining filter box behavior.